pkg/controller/install/certresources.go

```diff
@@ -253,8 +253,15 @@ func (i *StrategyDeploymentInstaller) installCertRequirementsForDeployment(deplo
 	ownerutil.AddNonBlockingOwner(service, i.owner)
 	service.SetLabels(map[string]string{OLMManagedLabelKey: OLMManagedLabelValue})
 
-	existingService, err := i.strategyClient.GetOpLister().CoreV1().ServiceLister().Services(i.owner.GetNamespace()).Get(service.GetName())
-	if err == nil {
+	// Optimistically create the Service
+	_, err := i.strategyClient.GetOpClient().CreateService(service)
+	if apierrors.IsAlreadyExists(err) {
+		// If the Service already exists, attempt to replace it if safe to do so.
```

Comment:
I know that this is building on top of existing logic to delete and then recreate, but is there a specific reason why it needs to be a delete followed by a create? Could the same goal be accomplished with a patch/update?

Reply:
IME this is because there are many rules for the service spec around updates and it's very easy to run into errors unless you delete then recreate them. I can't really say with certainty if this was or is a best-practice of any kind, just trying to give a potential reason why this code was set up this way.

Reply:
Yeah, that makes sense. This should definitely not block this PR since this is _generally_ the same logic. Was curious if there was any historical reasoning for this behavior that you might be aware of since I noticed there is also a `UpdateService()` method on the service client interface when combing through the code

Reply:
Server side apply will make sure you only need one call and will do the right thing (e.g. create if not exist) in case you want to consider it.